src/impl_constructors.rs

```diff
@@ -457,7 +460,7 @@ where
         // debug check for issues that indicates wrong use of this constructor
         debug_assert!(dimension::can_index_slice(&v, &dim, &strides).is_ok());
         ArrayBase {
-            ptr: nonnull_from_vec_data(&mut v),
+            ptr: nonnull_from_vec_data(&mut v).offset(head_ptr_offset(dim.slice(),strides.slice()).abs()),
```

Comment:
Why is there an abs here?

Reply:
When calculating the offset, the result is always less than or equal to zero (this can avoid using abs() when calculating the offset). Therefore, we need to use abs() to obtain a positive offset when getting the logical head pointer according to the actual head pointer in the memory.